### PR TITLE
feat: Initial changes to support OEP65

### PR DESCRIPTION
**Pull request by pedromartello**
**Description:**

This pull request introduces changes to support the use of [Piral](https://piral.io/) the micro-frontend framework used in the prototype for [OEP-65 - currently referred to OEPXXXX](https://github.com/openedx/open-edx-proposals/blob/426f6e09ffe615e77aa9205281d77012385a08d4/oeps/architectural-decisions/oep-XXXX-modular-micro-frontend-domains.rst).

Specific changes introduced: 

- Upgrade the `react-intl` dependency to `6.4.4`. The previous version `5.25.1` is incompatible with Piral and generated errors in the React component hierarchy.
- Upgrade `redux` to `8.1.1`. The previous version `7.1.1` is incompatible with Piral and generated errors where the Redux store was not available during component rendering.
- Refactored and exposed the internal function `mergeMessages`. Previously this method would overwrite existing messages with the new messages. The new version overloads this function to actually merge new messages with existing ones.
- Reordered initialization so that internationalization is loaded before authentication since authentication errors messages require  internationalization to be loaded.  
- Bumped version of `@edx/paragon` to `^21.1.2` to resolve dependency conflicts with the versions of `react-intl` and `react-redux` above

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @pedromartello!  Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


This is currently a draft pull request. When it is ready for our review and all tests are green, click "Ready for Review", or remove "WIP" from the title, as appropriate.

**Comment by e0d** [comment]
@pedromartello looks like there are some test failures.  I realize this is still draft, but wanted to mention.

**Comment by pedromartello** [comment]
> @pedromartello looks like there are some test failures. I realize this is still draft, but wanted to mention.

@e0d This is an npm dependency failure:  frontend-platform has a dependency on paragon which in turn declares a dependency on react-intl@5.25.1. The related change to paragon in this [PR:](https://github.com/openedx/paragon/pull/2525) has passed tests. I don't think I can declare the dependency as `5.25.1 || 6.4.4` as that would break downstream projects that depend on 6.4.4. 

Wondering if @arbrandes has a recommendation as he was looking at a related concern before I issued a PR for `openedx/frontend-app-shell`

**Comment by pedromartello** [comment]
@e0d @arbrandes - I moved this PR to "ready for review" as the required changes to paragon have been merged and released.

**Review by arbrandes — CHANGES_REQUESTED** [review]
No objections in principle, but I think it's best to separate out these changes into different commits:

*  One for the redux update, with a message linking to the 8.0.0 changelog ([this one](https://github.com/reduxjs/react-redux/releases?page=2)) noting that there are no (known) breaking changes
* Another for the react-intl upgrade, with a similar message
* A third one for the `mergeMessages()` change (or new function, depending)
* A fourth for the reordering of the `configureI18n` stanza, with an explanation.

These might have all been separate PRs, but I don't think we need to go that far unless somebody objects.

File: src/i18n/lib.js [hunk 1]
```diff
@@ -262,7 +265,7 @@ export function configure(options) {
   loggingService = options.loggingService;
   // eslint-disable-next-line prefer-destructuring
   config = options.config;
-  messages = Array.isArray(options.messages) ? mergeMessages(options.messages) : options.messages;
+  messages = Array.isArray(options.messages) ? merge({}, ...options.messages) : options.messages;
```
Comment (pedromartello) [inline src/i18n/lib.js:268]: @arbrandes - the only use of the mergeMessages function was here (and in the unit tests). Note how it has now been changed to use lodash.merge directly. The exports outside of /lib and the package itself were added in this commit.

File: src/i18n/lib.js [hunk 2]
```diff
@@ -236,12 +236,15 @@ const optionsShape = {
...
- * @param {Array} [messagesArray=[]]
+ * @param {Array} {Object} options.messages
  * @returns {Object}
  * @memberof module:Internationalization
  */
-export function mergeMessages(messagesArray = []) {
-  return Array.isArray(messagesArray) ? merge({}, ...messagesArray) : {};
+export function mergeMessages(newMessages) {
+  const msgs = Array.isArray(newMessages) ? merge({}, ...newMessages) : newMessages;
+  messages = merge(messages, msgs);
```
[the remark this answers is not in the document]
Comment (pedromartello) [reply src/i18n/lib.js:245]: see my comment below

**Comment by pedromartello** [comment]
Closing to follow the suggestions in this comment: https://github.com/openedx/frontend-platform/pull/535#pullrequestreview-1596147273

**Comment by openedx-webhooks** [comment]
@pedromartello Even though your pull request wasn’t merged, please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/frontend-platform&entry.867055334=https://github.com/openedx/frontend-platform/pull/535&entry.1484655318=https://github.com/pedromartello&entry.752974735=2023-08-10+22:11&entry.1917517419=2023-08-25+18:36&entry.2133058324=No) so we can improve your experience in the future.

**Pull request closed without merging**